File: client/tube.go
package main

import (
    "os"
    "encoding/json"
)

var TubeFile *os.File
var ResDecoder *json.Decoder
var FdTube uintptr

func InitTube(textTube uintptr, fdTube uintptr) {
    TubeFile = os.NewFile(textTube, "<das tube>")
    ResDecoder = json.NewDecoder(TubeFile)
    FdTube = fdTube
}

File: server/tube.c
#include "common.h"
#include "tube.h"

int TubeFd;
int FdTubeFd;
FILE *TubeFile;

void InitTubes(int textTube, int fdTube) {
    TubeFd = textTube;
    TubeFile = fdopen(textTube, "a+");
    if (!TubeFile) {
        Check_1("fdopen", -1);
    }
    FdTubeFd = fdTube;
}
